src/GenericCyclotomics.jl

```diff
@@ -381,8 +374,8 @@ function (R::GenericCycloRing)(f::Dict{UPolyFrac,UPoly}; simplify::Bool=true)  #
       else
         gp =
           evaluate(
-            numerator(g), [1], [substitute]
-          )//evaluate(denominator(g), [1], [substitute])
+            numerator(g), [1], [R.substitute]
+          )//evaluate(denominator(g), [1], [R.substitute])
```

Comment:
If the fraction was reduced (i.e. numerator and denomiator are coprime) *before* the substitution, then it still is after, correct? But the `//` method can't know that and thus may end up trying to reduce? (I don't know if it actually does that, I just wonder).

In any case, you next need numerator and denominator anyway. So perhaps something like this could be used in this loop instead:
```julia
  den_g = denominator(g)
  num_g = numerator(g)
  if R.congruence === nothing
    den_g = evaluate(den_g, [1], [substitute])
    num_g = evaluate(num_g, [1], [substitute])
  end
  a, r = divrem(num_g, den_g)
  push!(L, (c, den_g, r, a))
```
This would arguably also make the code a little bit nicer?

Reply:
> If the fraction was reduced (i.e. numerator and denomiator are coprime) before the substitution, then it still is after, correct?

No, not necessarily. This is kind of the whole point of this substitution. Take for example the fraction `(q+1)//2`. If `q` is now congruent to `1` modulo `2` then `R.substitute` is equal to `2*q+1`. After evaluation the fraction is `(2*q+2)//2` which is not reduced anymore. The idea is then that the fraction gets reduced to `q+1` which has a normal form of `0`. This detects that `exp(2*pi*i*(q+1)//2)` is equal to `1` for all `q` congruent to `1` modulo `2`.

Reply:
ah of course, gotcha.

So please add a comment explaining just that? :-)